### ISSUE TITLE
Not possible to create pretty JSON from headers.

### ISSUE THREAD

**Issue by carlbenson** [rabbitmq/rabbitmq-java-client#83]
`AMQP.BasicProperties.getHeaders()` returns a `Map<String, Object>` where the strings are of type `LongStringHelper.ByteArrayLongString` that is a private internal class which makes it impossible to write custom serializers for this class. Using Google Gson for instance, always inserts a `"bytes":` key before the value. While it is easy to write a serializer that converts the resulting `byte[]`, I'd like to have the string value only without the `"bytes":` key.

In short, how can I produce a regular Java string out of `LongStringHelper.ByteArrayLongString`?

**Comment by michaelklishin** [comment]
`ByteArrayLongString#toString`?

**Comment by carlbenson** [comment]
https://github.com/rabbitmq/rabbitmq-java-client/blob/master/src/com/rabbitmq/client/impl/LongStringHelper.java#L37

It's private.

**Comment by michaelklishin** [comment]
@carlbenson `toString` will be a part of the `LongString` interface in 3.6.0.

**Comment by michaelklishin** [comment]
@carlbenson for now, it should be possible to use `getBytes` (already part of the interface) and produce a string from that, using UTF-8 as encoding.
